langstream-api-gateway/src/main/java/ai/langstream/apigateway/websocket/handlers/AbstractHandler.java

```diff
@@ -189,16 +189,26 @@ private Gateway extractGateway(
         return selectedGateway;
     }
 
-    public GatewayRequestContext validateRequest(
+    public GatewayRequestContextImpl validateRequest(
             Map<String, String> pathVars,
             Map<String, String> queryString,
             Map<String, String> httpHeaders) {
         Map<String, String> options = new HashMap<>();
         Map<String, String> userParameters = new HashMap<>();
 
         final String credentials = queryString.remove("credentials");
+        final String adminCredentials = queryString.remove("admin-credentials");
+        final String adminCredentialsType = queryString.remove("admin-credentials-type");
+
+        final Map<String, String> adminCredentialsInputs = new HashMap<>();
 
         for (Map.Entry<String, String> entry : queryString.entrySet()) {
+            if (entry.getKey().startsWith("admin-credentials-input-")) {
```

Comment:
I am not sure we need this thing, it seems like a security hole, what's the use case ?